util/stmtsummary/statement_summary.go

```diff
@@ -388,20 +410,63 @@ func (ssMap *stmtSummaryByDigestMap) SetRefreshInterval(value string, inSession
 	}
 }
 
-func (ssMap *stmtSummaryByDigestMap) RefreshInterval() int64 {
+// refreshInterval gets the refresh interval for summaries.
+func (ssMap *stmtSummaryByDigestMap) refreshInterval() int64 {
 	return atomic.LoadInt64(&ssMap.sysVars.refreshInterval)
 }
 
+// SetHistorySize sets the history size for all summaries.
+func (ssMap *stmtSummaryByDigestMap) SetHistorySize(value string, inSession bool) {
+	ssMap.sysVars.Lock()
+	if inSession {
+		ssMap.sysVars.sessionHistorySize = value
+	} else {
+		ssMap.sysVars.globalHistorySize = value
+	}
+	sessionHistorySize := ssMap.sysVars.sessionHistorySize
+	globalHistorySize := ssMap.sysVars.globalHistorySize
+	ssMap.sysVars.Unlock()
+
+	// Calculate the cached `historySize`.
+	var size int
+	var err error
+	if ssMap.isSet(sessionHistorySize) {
+		size, err = strconv.Atoi(sessionHistorySize)
+		if err != nil {
+			size = 0
+		}
+	}
+	if size <= 0 {
+		size, err = strconv.Atoi(globalHistorySize)
+		if err != nil {
+			size = 0
+		}
+	}
+	if size > 0 {
```

Comment:
`historySize` will always not set to 0？

Reply:
after 
```sql
set @@session.tidb_stmt_summary_history_size=0;
set @@global.tidb_stmt_summary_history_size=0;
```
, the `historySize ` still not 0?

Reply:
Yes. Actually, history includes the current summary. So `historySize=1` indicates that there's only the current summary, and `historySize=0` makes no sense.
I designed it like this because I wish `xxx_history` would be enough to see all the summaries.
What do you think?

Reply:
Addressed. Now `historySize` is allowed to be 0.